File: solutions/N2_2023_F1_Contas_a_pagar.cpp
// OBI 2023 - Fase 1 - Nível 2: Contas a pagar (https://olimpiada.ic.unicamp.br/static/extras/obi2023/provas/ProvaOBI2023_f1p2.pdf)
#include <bits/stdc++.h>

using namespace std;

int solve(int V, int A, int F, int P)
{
    auto S = A + F + P;

    if (V >= S)
        return 3;
    else if (V >= A + F or V >= A + P or V >= F + P)
        return 2;
    else if (V >= A or V >= F or V >= P)
        return 1;

    return 0;
}

int main()
{
    int V, A, F, P;
    cin >> V >> A >> F >> P;

    cout << solve(V, A, F, P) << '\n';

    return 0;
}

File: solutions/N2_2023_F1_Contas_a_pagar.py
# NEPS 2321 - Contas a pagar (https://neps.academy/br/exercise/2321)

V = int(input())
A = int(input())
F = int(input())
P = int(input())

S = A + F + P

if V >= S:
    print(3)
elif V >= A + F or V >= A + P or V >= F + P:
    print(2)
elif V >= A or V >= F or V >= P:
    print(1)
else:
    print(0)
